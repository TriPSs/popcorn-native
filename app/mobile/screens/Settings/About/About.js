import React from 'react'
import PropTypes from 'prop-types'
import { View, Linking } from 'react-native'
import Updater from 'update-react-native-app'

import dimensions from 'modules/dimensions'
import i18n from 'modules/i18n'
import { navigate } from 'modules/RootNavigation'
import withIpFinder from 'modules/IpFinder/withIpFinder'

import Container from 'components/Container'
import Icon from 'components/Icon'
import Typography from 'components/Typography'
import TextButton from 'components/TextButton'

export const styles = {

  root: {
    paddingTop: dimensions.STATUSBAR_HEIGHT + dimensions.UNIT * 2,
  },

  title: {
    marginLeft: dimensions.UNIT,
    marginRight: dimensions.UNIT,
    marginBottom: dimensions.UNIT / 2,
  },

  container: {
    paddingTop: dimensions.UNIT / 2,
    paddingBottom: dimensions.UNIT,

    display: 'flex',
    flexDirection: 'row',
  },

  divider: {
    marginTop: 1,
  },

  iconContainer: {
    display: 'flex',
    justifyContent: 'center',
    margin: dimensions.UNIT,
  },
}

export const About = ({ data, ipFinder }) => (
  <Container style={styles.root}>

    <Typography
      style={styles.title}
      variant={'headline6'}>About</Typography>

    <Container
      style={styles.container}
      elevation={2}>

      <View style={styles.iconContainer}>
        <Icon name={'filmstrip'} />
      </View>

      <View>
        <Typography variant={'subtitle2'}>GraphQL-API</Typography>
        <Typography variant={'caption'}>
          {i18n.t('Version: {{version}}', { version: data?.status?.version ?? 'unknown' })}
        </Typography>
        <Typography variant={'caption'}>
          {i18n.t('Movies: {{movies}}', { movies: data?.status?.totalMovies ?? 'unknown' })}
        </Typography>
        <Typography variant={'caption'}>
          {i18n.t('Shows: {{shows}}', { shows: data?.status?.totalShows ?? 'unknown' })}
        </Typography>
      </View>
    </Container>

    <Container
      style={[styles.container, styles.divider]}
      elevation={2}>

      <View style={styles.iconContainer}>
        <Icon name={'database-import'} />
      </View>

      <View>
        <Typography variant={'subtitle2'}>Scraper</Typography>
        <Typography variant={'caption'}>
          {i18n.t('Version: {{version}}', { version: data?.scraper?.version ?? 'unknown' })}
        </Typography>
        <Typography variant={'caption'}>
          {i18n.t('Status: {{status}}', { status: data?.scraper?.status ?? 'unknown' })}
        </Typography>
        <Typography variant={'caption'}>
          {i18n.t('Uptime: {{uptime}}', { uptime: data?.scraper?.uptime ?? 'unknown' })}
        </Typography>
        <Typography variant={'caption'}>
          {i18n.t('Last updated: {{update}}', { update: data?.scraper?.updated ?? 'unknown' })}
        </Typography>
        <Typography variant={'caption'}>
          {i18n.t('Next update: {{update}}', { update: data?.scraper?.nextUpdate ?? 'unknown' })}
        </Typography>
      </View>
    </Container>

    <Container
      style={[styles.container, styles.divider]}
      elevation={2}>

      <View style={styles.iconContainer}>
        <Icon name={'popcorn'} />
      </View>

      <View style={styles.textContainer}>
        <Typography variant={'subtitle2'}>App</Typography>
        <Typography variant={'caption'}>
          {i18n.t('Version: {{version}}', { version: Updater?.UpdateRNApp?.versionName ?? 'unknown' })}
        </Typography>
      </View>
    </Container>

    <TextButton onPress={() => navigate('AppChangelog')}>
      Changelog
    </TextButton>

    <TextButton onPress={() => Linking.openURL(`http://${ipFinder.ip}:${ipFinder.SCRAPER_PORT}/files`)}>
      Log files
    </TextButton>

  </Container>
)

About.propTypes = {
  data: PropTypes.object,
}

About.defaultProps = {
  data: null,
}

export default withIpFinder(About)
