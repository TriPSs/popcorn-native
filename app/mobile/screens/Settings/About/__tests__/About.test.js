import React from 'react'
import { shallow } from 'enzyme'
import Updater  from 'update-react-native-app'

import { About } from '../About'

jest.mock('update-react-native-app')

describe('(Screens) Settings > About', () => {

  it('should render with everything unknown', () => {
    const result = shallow(
      <About />,
    )

    expect(result).toMatchSnapshot()
  })

  it('should render with everything unknown when data incorrect', () => {
    const result = shallow(
      <About
        data={{ fakeData: true }}
      />,
    )

    expect(result).toMatchSnapshot()
  })

  it('should render with app version', () => {
    Updater.UpdateRNApp = {
      versionName: 'test version'
    }

    const result = shallow(
      <About />,
    )

    expect(result).toMatchSnapshot()
  })

  it('should render', () => {
    const result = shallow(
      <About
        data={{
          status: {
            version: 'test',
            totalMovies: 1,
            totalShows: 1,
          },
          scraper: {
            version: 'test',
            uptime: 10,
            status: 'idle',
            updated: 'test',
            nextUpdate: 'test'
          },
        }}
      />,
    )

    expect(result).toMatchSnapshot()
  })


})
