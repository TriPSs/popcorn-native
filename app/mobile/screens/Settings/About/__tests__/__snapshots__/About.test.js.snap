// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Screens) Settings > About should render 1`] = `
<Container
  elevation={0}
  style={
    Object {
      "paddingTop": 16,
    }
  }
>
  <Typography
    color="white"
    component={null}
    emphasis="high"
    fontWeight="default"
    style={
      Object {
        "marginBottom": 4,
        "marginLeft": 8,
        "marginRight": 8,
      }
    }
    textProps={Object {}}
    variant="headline6"
  >
    About
  </Typography>
  <Container
    elevation={2}
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "paddingBottom": 8,
        "paddingTop": 4,
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
          "margin": 8,
        }
      }
    >
      <Icon
        color="white"
        emphasis="high"
        name="filmstrip"
        size={24}
        style={null}
      />
    </View>
    <View>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="subtitle2"
      >
        GraphQL-API
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Version: test
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Movies: 1
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Shows: 1
      </Typography>
    </View>
  </Container>
  <Container
    elevation={2}
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
          "paddingBottom": 8,
          "paddingTop": 4,
        },
        Object {
          "marginTop": 1,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
          "margin": 8,
        }
      }
    >
      <Icon
        color="white"
        emphasis="high"
        name="database-import"
        size={24}
        style={null}
      />
    </View>
    <View>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="subtitle2"
      >
        Scraper
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Version: test
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Status: idle
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Uptime: 10
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Last updated: test
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Next update: test
      </Typography>
    </View>
  </Container>
  <Container
    elevation={2}
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
          "paddingBottom": 8,
          "paddingTop": 4,
        },
        Object {
          "marginTop": 1,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
          "margin": 8,
        }
      }
    >
      <Icon
        color="white"
        emphasis="high"
        name="popcorn"
        size={24}
        style={null}
      />
    </View>
    <View>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="subtitle2"
      >
        App
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Version: test version
      </Typography>
    </View>
  </Container>
  <TextButton
    color="white"
    emphasis="high"
    onPress={[Function]}
    upperCase={true}
    variant="button"
  >
    Changelog
  </TextButton>
  <TextButton
    color="white"
    emphasis="high"
    onPress={[Function]}
    upperCase={true}
    variant="button"
  >
    Log files
  </TextButton>
</Container>
`;

exports[`(Screens) Settings > About should render with app version 1`] = `
<Container
  elevation={0}
  style={
    Object {
      "paddingTop": 16,
    }
  }
>
  <Typography
    color="white"
    component={null}
    emphasis="high"
    fontWeight="default"
    style={
      Object {
        "marginBottom": 4,
        "marginLeft": 8,
        "marginRight": 8,
      }
    }
    textProps={Object {}}
    variant="headline6"
  >
    About
  </Typography>
  <Container
    elevation={2}
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "paddingBottom": 8,
        "paddingTop": 4,
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
          "margin": 8,
        }
      }
    >
      <Icon
        color="white"
        emphasis="high"
        name="filmstrip"
        size={24}
        style={null}
      />
    </View>
    <View>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="subtitle2"
      >
        GraphQL-API
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Version: unknown
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Movies: unknown
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Shows: unknown
      </Typography>
    </View>
  </Container>
  <Container
    elevation={2}
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
          "paddingBottom": 8,
          "paddingTop": 4,
        },
        Object {
          "marginTop": 1,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
          "margin": 8,
        }
      }
    >
      <Icon
        color="white"
        emphasis="high"
        name="database-import"
        size={24}
        style={null}
      />
    </View>
    <View>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="subtitle2"
      >
        Scraper
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Version: unknown
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Status: unknown
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Uptime: unknown
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Last updated: unknown
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Next update: unknown
      </Typography>
    </View>
  </Container>
  <Container
    elevation={2}
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
          "paddingBottom": 8,
          "paddingTop": 4,
        },
        Object {
          "marginTop": 1,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
          "margin": 8,
        }
      }
    >
      <Icon
        color="white"
        emphasis="high"
        name="popcorn"
        size={24}
        style={null}
      />
    </View>
    <View>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="subtitle2"
      >
        App
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Version: test version
      </Typography>
    </View>
  </Container>
  <TextButton
    color="white"
    emphasis="high"
    onPress={[Function]}
    upperCase={true}
    variant="button"
  >
    Changelog
  </TextButton>
  <TextButton
    color="white"
    emphasis="high"
    onPress={[Function]}
    upperCase={true}
    variant="button"
  >
    Log files
  </TextButton>
</Container>
`;

exports[`(Screens) Settings > About should render with everything unknown 1`] = `
<Container
  elevation={0}
  style={
    Object {
      "paddingTop": 16,
    }
  }
>
  <Typography
    color="white"
    component={null}
    emphasis="high"
    fontWeight="default"
    style={
      Object {
        "marginBottom": 4,
        "marginLeft": 8,
        "marginRight": 8,
      }
    }
    textProps={Object {}}
    variant="headline6"
  >
    About
  </Typography>
  <Container
    elevation={2}
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "paddingBottom": 8,
        "paddingTop": 4,
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
          "margin": 8,
        }
      }
    >
      <Icon
        color="white"
        emphasis="high"
        name="filmstrip"
        size={24}
        style={null}
      />
    </View>
    <View>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="subtitle2"
      >
        GraphQL-API
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Version: unknown
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Movies: unknown
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Shows: unknown
      </Typography>
    </View>
  </Container>
  <Container
    elevation={2}
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
          "paddingBottom": 8,
          "paddingTop": 4,
        },
        Object {
          "marginTop": 1,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
          "margin": 8,
        }
      }
    >
      <Icon
        color="white"
        emphasis="high"
        name="database-import"
        size={24}
        style={null}
      />
    </View>
    <View>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="subtitle2"
      >
        Scraper
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Version: unknown
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Status: unknown
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Uptime: unknown
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Last updated: unknown
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Next update: unknown
      </Typography>
    </View>
  </Container>
  <Container
    elevation={2}
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
          "paddingBottom": 8,
          "paddingTop": 4,
        },
        Object {
          "marginTop": 1,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
          "margin": 8,
        }
      }
    >
      <Icon
        color="white"
        emphasis="high"
        name="popcorn"
        size={24}
        style={null}
      />
    </View>
    <View>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="subtitle2"
      >
        App
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Version: unknown
      </Typography>
    </View>
  </Container>
  <TextButton
    color="white"
    emphasis="high"
    onPress={[Function]}
    upperCase={true}
    variant="button"
  >
    Changelog
  </TextButton>
  <TextButton
    color="white"
    emphasis="high"
    onPress={[Function]}
    upperCase={true}
    variant="button"
  >
    Log files
  </TextButton>
</Container>
`;

exports[`(Screens) Settings > About should render with everything unknown when data incorrect 1`] = `
<Container
  elevation={0}
  style={
    Object {
      "paddingTop": 16,
    }
  }
>
  <Typography
    color="white"
    component={null}
    emphasis="high"
    fontWeight="default"
    style={
      Object {
        "marginBottom": 4,
        "marginLeft": 8,
        "marginRight": 8,
      }
    }
    textProps={Object {}}
    variant="headline6"
  >
    About
  </Typography>
  <Container
    elevation={2}
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "paddingBottom": 8,
        "paddingTop": 4,
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
          "margin": 8,
        }
      }
    >
      <Icon
        color="white"
        emphasis="high"
        name="filmstrip"
        size={24}
        style={null}
      />
    </View>
    <View>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="subtitle2"
      >
        GraphQL-API
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Version: unknown
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Movies: unknown
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Shows: unknown
      </Typography>
    </View>
  </Container>
  <Container
    elevation={2}
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
          "paddingBottom": 8,
          "paddingTop": 4,
        },
        Object {
          "marginTop": 1,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
          "margin": 8,
        }
      }
    >
      <Icon
        color="white"
        emphasis="high"
        name="database-import"
        size={24}
        style={null}
      />
    </View>
    <View>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="subtitle2"
      >
        Scraper
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Version: unknown
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Status: unknown
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Uptime: unknown
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Last updated: unknown
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Next update: unknown
      </Typography>
    </View>
  </Container>
  <Container
    elevation={2}
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
          "paddingBottom": 8,
          "paddingTop": 4,
        },
        Object {
          "marginTop": 1,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
          "margin": 8,
        }
      }
    >
      <Icon
        color="white"
        emphasis="high"
        name="popcorn"
        size={24}
        style={null}
      />
    </View>
    <View>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="subtitle2"
      >
        App
      </Typography>
      <Typography
        color="white"
        component={null}
        emphasis="high"
        fontWeight="default"
        style={null}
        textProps={Object {}}
        variant="caption"
      >
        Version: unknown
      </Typography>
    </View>
  </Container>
  <TextButton
    color="white"
    emphasis="high"
    onPress={[Function]}
    upperCase={true}
    variant="button"
  >
    Changelog
  </TextButton>
  <TextButton
    color="white"
    emphasis="high"
    onPress={[Function]}
    upperCase={true}
    variant="button"
  >
    Log files
  </TextButton>
</Container>
`;
